### DISCUSSION TITLE
Update h2 pin?

### DISCUSSION THREAD

**Discussion by HarrySky** [encode/httpx#1485] [General]
Hi, some time ago `h2` pin was updated in `httpcore`: https://github.com/encode/httpcore/pull/208

But it is still pinned to `3.*` in this package's `http2` extra:
https://github.com/encode/httpx/blob/0f280af8b170ed5cc48c12a894f71a8b5762f748/setup.py#L65

This is not an issue, as I can just add `h2>=3,<5` to my setup.py instead of using `httpx[http2]`, but maybe you want dependencies to be in sync with `httpcore`.

EDIT: Using git blame we can see that before `http2` extra - `httpcore[http2]` was used instead of `h2` dependency directly.

**Comment by florimondmanca** [comment]
Thanks, that's interesting.

It seems the direct fix would be to bump h2 in HTTPX as well (we don't have dependabot here so dependencies don't get bumped automatically).

I wonder what would happen if we had the `http2` extra refer to `httpcore[http2]`, though?
